### PR TITLE
[PSA-5] circumvent the flow session for self issue of membership

### PR DESCRIPTION
**Pull request by aelahi23**
1) Updates the status (passed)
2) Check only BNO can call the flow(passed)

### CODE REVIEW

File: bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt [hunk 1]
```diff
@@ -2,60 +2,106 @@ package com.r3.businessnetworks.membership.flows.bno
...
+ * @param bnoMembership bno membership state to be activated
  */
+@Suppress("UNUSED_VARIABLE")
 @StartableByRPC
 @InitiatingFlow(version = 2)
-open class SelfIssueMembershipFlow(val metaData : Any) : FlowLogic<SignedTransaction>() {
+open class SelfIssueMembershipFlow(val bnoMembership: StateAndRef<MembershipState<Any>>) : FlowLogic<SignedTransaction>() {
     companion object {
-        object RequestingMembership : ProgressTracker.Step("Requesting Membership")
+        object ActivatedMembership : ProgressTracker.Step("Membership Activated")
```
Comment (ivanr3) [inline bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: Static constants are usually named in upper case, i.e. ACTIVATING_MEMBERSHIP

File: bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt [hunk 2]
```diff
@@ -2,60 +2,106 @@ package com.r3.businessnetworks.membership.flows.bno
...
+        // stop if Node is not a BNO
         throwExceptionIfNotBNO(ourIdentity, serviceHub)
+        val configuration = serviceHub.cordaService(BNOConfigurationService::class.java)
+        val notary = configuration.notaryParty()
+        logger.info("Membership is being activated")
+        progressTracker.currentStep = ActivatingMembership
+
+
+            val txBuilder = TransactionBuilder(notary)
+                .addInputState(bnoMembership)
```
Comment (ivanr3) [inline bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: Is this flow meant to do membership issuance? If yes - then where is the existing membership coming from?

File: bn-apps/memberships-management/membership-service/src/test/kotlin/com/r3/businessnetworks/membership/flows/SelfIssueMembershipFlowTest.kt [hunk 3]
```diff
@@ -1,51 +1,54 @@
...
+import net.corda.core.utilities.getOrThrow
 import org.junit.Test
-import kotlin.test.assertEquals
-import kotlin.test.assertTrue
 import kotlin.test.fail
 
-class SelfIssueMembershipFlowTest : AbstractFlowTest(numberOfBusinessNetworks = 2,
-        numberOfParticipants = 2,
-        participantRespondingFlows = listOf(RequestMembershipFlowResponder::class.java)) {
+class SelfIssueMembershipFlow : AbstractFlowTest(
```
Comment (ivanr3) [inline bn-apps/memberships-management/membership-service/src/test/kotlin/com/r3/businessnetworks/membership/flows/SelfIssueMembershipFlowTest.kt]: Please name tests with "Test" in the end, i.e. SelftIssueMembershipFLowTest

File: bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt (hunk 2 above)
Comment (aelahi23) [reply bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: (line 27) SelfIssueMembershipFlow(val bnoMembership: StateAndRef<MembershipState<Any>>) the parameter bnoMembership will reference the existing bno Member to change it's status

**Comment by aelahi23** [comment]
> Hi Arif. Please see comments inline. Also what about amending and revoking memberships? Would it still work if BNO and member have the same identity?

**Pull request closed without merging**